### PR TITLE
IA-3726 Front leo update for supporting start/stop Azure runtime

### PR DESCRIPTION
**Pull request by Qi77Qi**
Tested locally via swagger

```
{"@timestamp":"2022-09-19T19:06:35.311Z","@version":"1","logger_name":"org.broadinstitute.dsde.workbench.leonardo.http.Boot","traceId":"56fc1e18-f38a-49c5-9525-cbef858fa13f","version":"8b9f5c2b6","thread_name":"io-compute-5","severity":"INFO","serviceContext":{"service":"leonardo","version":"8b9f5c2b6"},"message":"Subscriber Successfully received data: \"{\\\"messageType\\\":\\\"startRuntime\\\",\\\"runtimeId\\\":193,\\\"traceId\\\":\\\"56fc1e18-f38a-49c5-9525-cbef858fa13f\\\"}\"\nattributes {\n  key: \"leonardo\"\n  value: \"true\"\n}\nattributes {\n  key: \"traceId\"\n  value: \"56fc1e18-f38a-49c5-9525-cbef858fa13f\"\n}\nmessage_id: \"5628873310099561\"\npublish_time {\n  seconds: 1663614395\n  nanos: 371000000\n}\n."}
{"@timestamp":"2022-09-19T19:06:35.311Z","@version":"1","logger_name":"org.broadinstitute.dsde.workbench.leonardo.http.Boot","duration":"351","result":"Succeeded","googleCall":"Publishing data: \"{\\\"messageType\\\":\\\"startRuntime\\\",\\\"runtimeId\\\":193,\\\"traceId\\\":\\\"56fc1e18-f38a-49c5-9525-cbef858fa13f\\\"}\"\nattributes {\n  key: \"leonardo\"\n  value: \"true\"\n}\nattributes {\n  key: \"traceId\"\n  value: \"56fc1e18-f38a-49c5-9525-cbef858fa13f\"\n}\n","traceId":"56fc1e18-f38a-49c5-9525-cbef858fa13f","version":"8b9f5c2b6","thread_name":"io-compute-6","severity":"INFO","serviceContext":{"service":"leonardo","version":"8b9f5c2b6"},"message":{"response":"()","result":"Succeeded"}}
{"@timestamp":"2022-09-19T19:06:36.762Z","@version":"1","logger_name":"org.broadinstitute.dsde.workbench.leonardo.http.Boot","traceId":"56fc1e18-f38a-49c5-9525-cbef858fa13f","version":"8b9f5c2b6","thread_name":"io-compute-3","stack_trace":"java.lang.RuntimeException: init bucket not found for Azure/fad90753-2022-4456-9b0a-c7e5b934e408/3efc5bdf-be0e-44e7-b1d7-c08931e3c16c/mrg-terra-workspace-20220412104730/saturn-c9a9e1a4-a335-48c8-85fd-2d50fa5919a0 in DB\n\tat org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessageSubscriber.$anonfun$handleStartRuntimeMessage$10(LeoPubsubMessageSubscriber.scala:383)\n\tat cats.ApplicativeError.fromOption(ApplicativeError.scala:318)\n\tat cats.ApplicativeError.fromOption$(ApplicativeError.scala:315)\n\tat flatMap @ org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessageSubscriber.$anonfun$handleStartRuntimeMessage$8(LeoPubsubMessageSubscriber.scala:382)\n\tat modify @ fs2.internal.Scope.close(Scope.scala:262)\n\tat flatMap @ fs2.Compiler$Target.flatMap(Compiler.scala:163)\n\tat as @ org.broadinstitute.dsde.workbench.leonardo.http.Boot$.run(Boot.scala:710)\n\tat delay @ org.broadinstitute.dsde.workbench.leonardo.db.DbRef.$anonfun$inTransaction$1(DbReference.scala:96)\n\tat fromFuture @ org.broadinstitute.dsde.workbench.leonardo.db.DbRef.$anonfun$inTransaction$1(DbReference.scala:96)\n","severity":"ERROR","serviceContext":{"service":"leonardo","version":"8b9f5c2b6"},"message":"Fail to process pubsub message due to unexpected error"}
```

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by Qi77Qi** [comment]
jenkins retest

File: http/src/main/resources/swagger/api-docs.yaml [hunk 1]
```diff
@@ -632,6 +632,84 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/ErrorReport"
+  "/api/v2/runtimes/{workspaceId}/{name}/start":
+    post:
+      summary: Start runtime (WIP).
+      description: >
+        Returns information about an existing azure runtime managed by Leo.
+        Poll this to find out when your runtime has finished starting up.
```
Comment (n8landolt) [inline http/src/main/resources/swagger/api-docs.yaml]: What does the polling look like?

Comment (Qi77Qi) [reply http/src/main/resources/swagger/api-docs.yaml]: oh, that's probably copy-paste error...will clean up

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala [hunk 2]
```diff
@@ -282,6 +262,59 @@ class RuntimeV2ServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
...
+      .raiseError[Unit](
+        RuntimeNotFoundException(runtime.cloudContext, runtimeName, "permission denied", Some(ctx.traceId))
+      )
+      .whenA(!hasPermission)
+    _ <-
+      if (runtime.status.isStartable) F.unit
+      else
+        F.raiseError[Unit](RuntimeCannotBeStartedException(runtime.cloudContext, runtime.runtimeName, runtime.status))
+    _ <- clusterQuery.updateClusterStatus(runtime.id, RuntimeStatus.PreStarting, ctx.now).transaction
+    _ <- publisherQueue.offer(StartRuntimeMessage(runtime.id, Some(ctx.traceId)))
```
Comment (n8landolt) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala:288]: If I'm reading this correctly, 287 and 288 should be on line 284

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala [hunk 3]
```diff
@@ -282,6 +262,59 @@ class RuntimeV2ServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
...
+      .raiseError[Unit](
+        RuntimeNotFoundException(runtime.cloudContext, runtimeName, "permission denied", Some(ctx.traceId))
+      )
+      .whenA(!hasPermission)
+    _ <-
+      if (runtime.status.isStoppable) F.unit
+      else
+        F.raiseError[Unit](RuntimeCannotBeStoppedException(runtime.cloudContext, runtime.runtimeName, runtime.status))
+    _ <- clusterQuery.updateClusterStatus(runtime.id, RuntimeStatus.PreStopping, ctx.now).transaction
+    _ <- publisherQueue.offer(StopRuntimeMessage(runtime.id, Some(ctx.traceId)))
```
Comment (n8landolt) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala:315]: Same comment for startRuntime

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala (hunk 2 above)
Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala:288]: why do you think that way?

Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala:288]: oh I guess you are thinking of sth like
```
if (condition)
 do X
else
 do y
```

But it's quite common in Scala, we do
```
for {
 _ <- someCheck, and if this fails, terminate the flow
 _ <- happy path flow
} yield ()
```

Comment (n8landolt) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala:288]: I misread the indentation - I thought 287 and 288 were part of the error 'else' path.

It looks good! And the tests check for the message as well.

**Review by n8landolt — APPROVED** [review]
Looks good to me!

**Pull request merged**